src/workerd/api/node/tests/timers-nodejs-test.js

```diff
@@ -155,3 +156,343 @@ export const testRefresh = {
     await promise;
   },
 };
+
+// Tests are taken from
+// https://github.com/nodejs/node/blob/7bc2946293757389468f1fa09714860f8e1147b7/test/parallel/test-timers-immediate-promisified.js
+export const timersImmediatePromise = {
+  async test() {
+    {
+      const ac = new AbortController();
+      const signal = ac.signal;
+      const { promise, resolve } = Promise.withResolvers();
+      rejects(timersPromises.setImmediate(10, { signal }), /AbortError/).then(
```

Comment:
Would this work? Might be a bit nicer.

```
queueMicrotask(() => ac.abort());
await rejects(timersPromises.setImmediate(10, { signal }));
```

Or perhaps

```
await rejects(timersPromises.setImmediate(10, { signal: AbortSignal.timeout(1) });
```

Reply:
Yeah, I just converted existing Node.js tests.

Reply:
This update broke ASAN tests